src/main/java/seedu/address/logic/commands/ListTransactionCommand.java

```diff
@@ -33,14 +33,19 @@ public ListTransactionCommand(Index index) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        if (model.getIsViewTransactions()) {
+            throw new CommandException(String.format(Messages.MESSAGE_MUST_BE_PERSON_LIST, COMMAND_WORD));
+        }
+
         List<Person> lastShownList = model.getFilteredPersonList();
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
 
         Person selected = lastShownList.get(index.getZeroBased());
         model.updateFilteredPersonList(new IsSelectedPredicate(model, index));
-        model.setViewTransactions(true);
+        //model.setViewTransactions(true);
```

Comment:
Same as above

Reply:
i have moved that logic to model#updateTransactionList()